### PR TITLE
Fix stupid mistakes for "very high CA" mode

### PR DESCRIPTION
**Pull request by rawfiner**
This mode actually worked pretty well considering the stupid mistakes I made...
The side effect was that the image was brightened, and was especially visible on blue sky.
Should be better now! :-)

### CODE REVIEW

File: src/iop/cacorrectrgb.c [hunk 1]
```diff
@@ -399,23 +399,35 @@ dt_omp_firstprivate(in, blurred_in, manifold_lower, manifold_higher, width, heig
...
           const float pixel = fmaxf(in[k * 4 + c], 1E-6f);
-          const float log_diff = logf(pixel) - pixelg;
+          const float log_diff = log2f(pixel) - pixelg;
           manifold_higher[k * 4 + c] = log_diff * w;
         }
         manifold_higher[k * 4 + guide] = fmaxf(in[k * 4 + guide], 0.0f) * w;
         manifold_higher[k * 4 + 3] = w;
+        // manifold_lower still contains the values from first iteration
+        // -> reset it.
+        for(size_t c = 0; c < 4; c++)
```
Comment (TurboGit) [inline src/iop/cacorrectrgb.c:409]: you should be able to use: for_four_channels()

it is a macro that will vectorize this loop if possible.

File: src/iop/cacorrectrgb.c [hunk 2]
```diff
@@ -399,23 +399,35 @@ dt_omp_firstprivate(in, blurred_in, manifold_lower, manifold_higher, width, heig
...
           const float pixel = fmaxf(in[k * 4 + c], 1E-6f);
-          const float log_diff = logf(pixel) - pixelg;
+          const float log_diff = log2f(pixel) - pixelg;
           manifold_lower[k * 4 + c] = log_diff * w;
         }
         manifold_lower[k * 4 + guide] = fmaxf(in[k * 4 + guide], 0.0f) * w;
         manifold_lower[k * 4 + 3] = w;
+        // manifold_higher still contains the values from first iteration
+        // -> reset it.
+        for(size_t c = 0; c < 4; c++)
```
Comment (TurboGit) [inline src/iop/cacorrectrgb.c:427]: likewise

**Review by TurboGit — CHANGES_REQUESTED** [review]
The vectorization should work fine as you are using aligned memory. But instead of using `dt_alloc_sse_ps()` please use `dt_alloc_align_float() `it is equivalent but better avoid using sse here as it is not an sse code path.

File: src/iop/cacorrectrgb.c (hunk 1 above)
Comment (TurboGit) [reply src/iop/cacorrectrgb.c:409]: EDIT: not possible as the loop is already inside a omp/simd section.

File: src/iop/cacorrectrgb.c (hunk 2 above)
Comment (TurboGit) [reply src/iop/cacorrectrgb.c:427]: EDIT: not possible as the loop is already inside a omp/simd section.

**Comment by TurboGit** [comment]
@rawfiner : so the only needed changes is to `dt_alloc_sse_ps()`.

**Comment by rawfiner** [comment]
I pushed the change, thank you for the review @TurboGit !

**Review by TurboGit — APPROVED** [review]
Looks good, thanks !

**Pull request merged**